### PR TITLE
DM-44047: Update bib generation code

### PR DESCRIPTION
**Pull request by timj**
* Use pybtex everywhere to ensure consistency.
* Sort entries on write to minimize diffs.
* Add --external option to command-line to allow the static entries and previous output to be read in and used as the baseline.

The latter option prevents documents from disappearing from the bib file if they temporarily were dropped by Algolia.

### CODE REVIEW

**Comment by timj** [comment]
Finally got all the tests working. There is a pybtex bug that messed everything up.

File: bin/generateBibfile.py [hunk 1]
```diff
@@ -70,14 +97,47 @@ def generate_bibfile(outfile, query=""):
...
+    # will complain if it finds Document-123 and document-123 in the dict.
+    all_data: BibDict[str, pybtex.database.Entry] = BibDict()
+    if external:
+        for bibfile in external:
+            with open(bibfile) as fd:
+                this_bib = BibliographyData.from_string(fd.read(), "bibtex")
+            all_data.update(this_bib.entries)
+
+    # Overwrite the entries from the search.
+    all_data.update(search_data.entries)
```
Comment (womullan) [inline bin/generateBibfile.py:116]: update updates and adds new entries ?

Comment (womullan) [reply bin/generateBibfile.py:116]: ok i see it does

**Review by womullan — APPROVED** [review]
ok

**Pull request merged**